-1 
Tested at: 81b17665fccde80fcc62a191a6e7e5e4db060ec8
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc493/cms/cmssw/CMSSW_8_1_X_2016-03-09-2300/src/Geometry/CommonDetUnit/interface/GeomDet.h:157:16: note: overridden virtual function is here
  virtual void setSurfaceDeformation(const SurfaceDeformation * deformation);
               ^
2 warnings generated.
2 warnings generated.
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-03-09-2300/src/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc:41:14: error: default initialization of an object of const type 'const (anonymous namespace)::Count' without a user-provided default constructor
 const Count count;
             ^
                  {}
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-03-09-2300/src/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc:294:78: warning: taking the absolute value of unsigned type 'unsigned long' has no effect [-Wabsolute-value]
        localAmplitudes.at( affectedStrip ) += chargeDepositedOnStrip * coupling.at(abs( affectedStrip - strip )) ;
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13672/11815/summary.html

--- a/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc
+++ b/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc
@@ -13,6 +13,7 @@
 
 namespace {
   struct Count {
+    Count() {}
 #ifdef SISTRIP_COUNT
     // note: this code is not thread safe, counts will be inaccurate if run with multiple threads
     double ncall=0;

--- a/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc
+++ b/SimTracker/SiStripDigitizer/plugins/SiTrivialInduceChargeOnStrips.cc
@@ -30,15 +30,17 @@ namespace {
       std::cout << "strips  " << nstr/ndep << " " << std::sqrt(nstr2*ndep -nstr*nstr)/ndep << std::endl;
       std::cout << "vaules  " << ncv << " " << maxv << " " << nval/ncv << " " << std::sqrt(nval2*ncv -nval*nval)/ncv << std::endl;
     }
-#else
-    void dep(double) {}
-    void str(double) {}
-    void val(double) {}
-    void zero() {}    
-#endif
   };
-  
  Count count;
+#else
+    void dep(double) const {}
+    void str(double) const {}
+    void val(double) const {}
+    void zero() const {}    
+ };
+ const Count count;
+#endif
+  
 }
 
 

